src/NUnitFramework/tests/SolutionTests/NuspecDependenciesTests.cs

```diff
@@ -0,0 +1,283 @@
+// Copyright (c) Charlie Poole, Rob Prouse and Contributors. MIT License - see LICENSE.txt
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+
+namespace NUnit.Framework.Tests.SolutionTests
+{
+    /// <summary>
+    /// Checks that the csproj of the framework projects are reflected correctly in the nuspec files.
+    /// </summary>
+    [TestFixture("framework/nunit.nuspec", "framework/nunit.framework.csproj")]
+    [TestFixture("nunitlite/nunitlite.nuspec", "nunitlite/nunitlite.csproj")]
+    internal sealed class NuspecDependenciesTests
+    {
+        private readonly string _nuspecPath;
+        private readonly string _csprojPath;
+        private readonly string _propsPath;
+
+        private Dictionary<string, List<PackageWithVersion>> _nuspecPackages;
+        private Dictionary<string, List<PackageWithVersion>> _csprojPackages;
+        private Dictionary<string, string> _csprojPackageVersions;
+
+        private const string Root = "../../../../../../";
+        private const string NotSpecified = nameof(NotSpecified);
+
+        public NuspecDependenciesTests(string nuspecPath, string csProjPath)
+        {
+            _nuspecPath = Path.GetFullPath($"{Root}/nuget/{nuspecPath}");
+            _csprojPath = Path.GetFullPath($"{Root}/src/NUnitFramework/{csProjPath}");
+            _propsPath = Path.GetFullPath($"{Root}/src/NUnitFramework/Directory.Packages.props");
+        }
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            _nuspecPackages = NuspecReader.ExtractNuspecPackages(_nuspecPath);
+            _csprojPackageVersions = PackageVersionReader.ExtractPackageVersions(_propsPath);
+            _csprojPackages = CsprojReader.ExtractCsprojPackages(_csprojPath, _csprojPackageVersions);
+        }
+
+        [Test]
+        public void NUnitFrameworkNuspecContainsCorrectDependencies()
```

Comment:
```suggestion
        public void AllPackagesInCsprojIsInNuspec()
```

nitpick: Given we do the inverse test below named `AllPackagesInNuspecIsInCsproj`, it may be clearer to name this test in a similar way